flake.nix

```diff
@@ -16,10 +20,27 @@
         inputs.haskell-flake.flakeModule
         inputs.treefmt-flake.flakeModule
         inputs.check-flake.flakeModule
+        # TODO: Is there an idiomatic way to specify custom overlays for `pkgs`
+        # passed by flake-parts (rather than hacking around it like we do by
+        # setting _module.args here)?
+        ({
+          perSystem = { inputs', system, ... }: {
+            _module.args.pkgs = import nixpkgs {
+              inherit system;
+              overlays = [
+                (final: prev: {
+                  all-cabal-hashes = inputs.all-cabal-hashes;
+                })
+              ];
+            };
+          };
+        })
```

Comment:
@roberth Is there a more idiomatic way to do this with `flake-parts`? Assuming @locallycompact's `horizon-platform` will export this particular overlay.

Reply:
Ideally a flake shouldn't have to add an overlay to its nixpkgs dependency.
OTOH having to support this seems inevitable. We could add an optional module to flake-parts for the purpose of initializing `pkgs`. Something like NixOS' `misc/nixpkgs.nix` but without the legacy stuff; just `hostPlatform`, `buildPlatform`, `overlays`, `config`.

Reply:
I think in Haskell it's basically impossible to avoid. You can't have two package exports close over a different version of the same dependency so the entire build plan needs to be injected at the point of use.